be/src/block_cache/block_cache.cpp

```diff
@@ -66,6 +66,9 @@ Status BlockCache::init(const CacheOptions& options) {
 
 Status BlockCache::write_buffer(const CacheKey& cache_key, off_t offset, const IOBuffer& buffer,
                                 WriteCacheOptions* options) {
+    if (!_kv_cache) {
```

Comment:
Too many If

Reply:
> Too many If

Changed it to check before calling datacache functions.